contrib/olivere/elastic/elastictrace.go

```diff
@@ -48,15 +49,12 @@ func (t *httpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	)
 	defer span.Finish()
 
-	contentLength, err := strconv.Atoi(req.Header.Get("Content-Length"))
-	if req.Body != nil && err != nil && contentLength < maxContentLength {
-		buf, err := ioutil.ReadAll(req.Body)
-		req.Body.Close()
-		if err != nil {
-			return nil, err
+	if req.Body != nil {
+		buf := bufio.NewReaderSize(req.Body, maxContentLength)
```

Comment:
Looks great, last two things:
- Let's call this `r` instead of `buf` please. It's a more standard name for a reader.
- Can we please do the same on the other `ioutil.ReadAll` call further down? To avoid this thing happening over there as well.